src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/ValidateApiAction.java

```diff
@@ -0,0 +1,144 @@
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.ActionGroupsV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.ConfigV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.InternalUserV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleMappingsV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.TenantV7;
+import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
+
+public class ValidateApiAction extends AbstractApiAction {
+
+    @Inject
+    public ValidateApiAction(final Settings settings, final Path configPath, final RestController controller, final Client client,
+                             final AdminDNs adminDNs, final ConfigurationRepository cl, final ClusterService cs, final PrincipalExtractor principalExtractor,
+                             final PrivilegesEvaluator evaluator, ThreadPool threadPool, AuditLog auditLog) {
+        super(settings, configPath, controller, client, adminDNs, cl, cs, principalExtractor, evaluator, threadPool, auditLog);
+    }
+
+    @Override
+    protected void registerHandlers(RestController controller, Settings settings) {
+        controller.registerHandler(Method.GET, "/_opendistro/_security/api/validate", this);
+    }
+
+    @Override
+    protected Endpoint getEndpoint() {
+        return Endpoint.VALIDATE;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected void handleGet(RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException {
+
+        final boolean acceptInvalid = request.paramAsBoolean("accept_invalid", false);
+
+        final SecurityDynamicConfiguration<?> loadedConfig = load(CType.CONFIG, true, acceptInvalid);
+
+        if (loadedConfig.getVersion() != 1) {
+            badRequestResponse(channel, "Can not migrate configuration because it was already migrated.");
+            return;
+        }
+
+        try {
+            final SecurityDynamicConfiguration<ConfigV6> configV6 = (SecurityDynamicConfiguration<ConfigV6>) loadedConfig;
+            final SecurityDynamicConfiguration<ActionGroupsV6> actionGroupsV6 = (SecurityDynamicConfiguration<ActionGroupsV6>) load(CType.ACTIONGROUPS, true, acceptInvalid);
+            final SecurityDynamicConfiguration<InternalUserV6> internalUsersV6 = (SecurityDynamicConfiguration<InternalUserV6>) load(CType.INTERNALUSERS, true, acceptInvalid);
+            final SecurityDynamicConfiguration<RoleV6> rolesV6 = (SecurityDynamicConfiguration<RoleV6>) load(CType.ROLES, true, acceptInvalid);
+            final SecurityDynamicConfiguration<RoleMappingsV6> rolesmappingV6 = (SecurityDynamicConfiguration<RoleMappingsV6>) load(CType.ROLESMAPPING, true, acceptInvalid);
```

Comment:
Could NodesDn also be added to validation API ?

Reply:
NodesDn doesn;t have different formatting for 6.x or 7.x 
there is no (NodesDnV6 or NodesDnV7) 

Do we still need validation for same formatting?

Reply:
The migration step adds the meta data right and this validation api ensures it was able to do so. Hence I thought it will be useful to add it here. 

Also it is updated in opendistro-1.1 branch https://github.com/opendistro-for-elasticsearch/security/blob/opendistro-1.1/src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/ValidateApiAction.java#L103

Reply:
Migration steps requires these steps .. because migration steps will export current config from index and re-create index from it.. 

Validation is pre-migration steps to validate if config can be migrated to 7.x version or not.. (Validation API was just build to cover any corner cases for invalid configs)
Post validation, we can migrate where we actually export current config, re-format it and re-create index from it.. 

Regarding, presence in 1.1 branch is: Dn hot reload initially built with two different classes one for 6.x and one 7.x eventhough format for both versions were same.  And it was removed before merging as there was no sense of keeping two different classes with same format. 

I can still add validation part to stay in sync with 1.1 branch, but it is simply not required. Or on other side we remove it from 1.1 branch too.

Reply:
Sounds good as the the NodeDn structure is the same!